cumulusci/tasks/salesforce/sobject_preflights.py

```diff
@@ -0,0 +1,78 @@
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import TaskOptionsError
+
+from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+
+class CheckSobjectOWDs(BaseSalesforceApiTask):
+    task_options = {
+        "org_wide_defaults": {
+            "description": "The Organization-Wide Defaults to check, "
+            "organized as a list with each element containing the keys api_name, "
+            "internal_sharing_model, and external_sharing_model. NOTE: you must have "
+            "External Sharing Model turned on in Sharing Settings to use the latter feature. "
+            "Checking External Sharing Model when it is turned off will fail the preflight.",
+            "required": True,
+        }
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        if "org_wide_defaults" not in self.options:
+            raise TaskOptionsError("org_wide_defaults is a required option")
+
+        if not all("api_name" in entry for entry in self.options["org_wide_defaults"]):
+            raise TaskOptionsError("api_name must be included in each entry")
+
+        if not all(
+            "internal_sharing_model" in entry or "external_sharing_model" in entry
+            for entry in self.options["org_wide_defaults"]
+        ):
+            raise TaskOptionsError("Each entry must include a sharing model to check.")
+
+        self.owds = {
+            entry["api_name"]: (
+                entry.get("internal_sharing_model"),
+                entry.get("external_sharing_model"),
+            )
+            for entry in self.options["org_wide_defaults"]
+        }
+
+    def _check_owds(self, sobject, result):
+        internal = (
+            result["InternalSharingModel"] == self.owds[sobject][0]
+            if self.owds[sobject][0]
+            else True
+        )
+        external = (
+            result["ExternalSharingModel"] == self.owds[sobject][1]
+            if self.owds[sobject][1]
+            else True
+        )
+        return internal and external
+
+    def _run_task(self):
+        try:
+            ext = (
+                ", ExternalSharingModel"
+                if any(owd[1] is not None for owd in self.owds.values())
+                else ""
+            )
+            self.return_values = all(
+                self._check_owds(
+                    obj,
+                    self.sf.query(
+                        f"SELECT InternalSharingModel{ext} "
+                        "FROM EntityDefinition "
+                        f"WHERE QualifiedApiName = '{obj}'"
+                    )["records"][0],
```

Comment:
Is it possible to do a single query for all the api names?

Reply:
Seems you can! The Tooling API is often fussy about multi-row queries, but it works on this entity.